features/include/pcl/features/good.h

```diff
@@ -0,0 +1,385 @@
+
+      /** \brief sign disambiguation of axis X
+ 	* \param[in] XoZ_projected_view pointer to a point cloud.
+	* \param[in] threshold used to deal with the special case when a point is close to the YoZ plane
+	* \param[out] sign the resultant sign (either 1 or -1) 
+        */       
+      void
+      signDisambiguationXAxis (PointCloudIn  XoZ_projected_view, float threshold, int &sign );
+      
+      /** \brief sign disambiguation of axis Y
+ 	* \param[in] YoZ_projected_view pointer to a point cloud.
+	* \param[in] threshold used to deal with the special case when a point is close to the XoZ plane
+	* \param[out] sign the resultant sign (either 1 or -1) 
+        */            
+      void 
+      signDisambiguationYAxis (PointCloudIn  YoZ_projected_view, float threshold, int &sign );
+
+      /** \brief create a 2D histogram from YOZ projection
+ 	* \param[in] YoZ_projected_view pointer to a point cloud.
+	* \param[in] largest_side largest side of object bounding box
+	* \param[in] number_of_bins  number of bins along one dimension;
+	* \param[in] sign either 1 or -1
+	* \param[out] YOZ_histogram a 2D vector of unsigned int
+        */            
+      void
+      create2DHistogramFromYOZProjection (PointCloudIn  YOZ_projected_view, double largest_side, unsigned int number_of_bins, 
+					   int sign, std::vector<std::vector<unsigned int> > &YOZ_histogram);
+
+      /** \brief create a 2D histogram from XOZ projection
+ 	* \param[in] XoZ_projected_view pointer to a point cloud.
+	* \param[in] largest_side largest side of object bounding box.
+	* \param[in] number_of_bins  number of bins along one dimension.
+	* \param[in] sign either 1 or -1.
+	* \param[out] XOZ_histogram a 2D vector of unsigned int.
+        */            
+      void 
+      create2DHistogramFromXOZProjection ( PointCloudIn  XOZ_projected_view, double largest_side, 
+					    unsigned int number_of_bins, int sign, std::vector < std::vector<unsigned int> > &XOZ_histogram);
+      
+      /** \brief create a 2D histogram from XOY projection
+ 	* \param[in] XoY_projected_view pointer to a point cloud.
+	* \param[in] largest_side largest side of object bounding box.
+	* \param[in] number_of_bins  number of bins along one dimension.
+	* \param[in] sign either 1 or -1.
+	* \param[out] XOY_histogram a 2D vector of unsigned int.
+        */  
+      void 
+      create2DHistogramFromXOYProjection (PointCloudIn  XOY_projected_view,	double largest_side,
+					unsigned int number_of_bins, int sign, std::vector < std::vector<unsigned int> > &XOY_histogram);
+
+      /** \brief normalizing a 1D histogram
+ 	* \param[in] histogram 1D histogram (int).
+	* \param[out] normalized_histogram the resultant normalized_histogram (float).
+        */       
+      void
+      normalizingHistogram (std::vector <unsigned int> histogram, std::vector <float> &normalized_histogram);

```

Comment:
Pass `histogram` by const reference to avoid unnecessary copies.
Further, please use verbs for function names, i.e. `normalizeHistogram`.